app/org/sagebionetworks/bridge/services/AccountWorkflowService.java

```diff
@@ -203,16 +238,28 @@ private void sendPasswordResetRelatedEmail(Study study, String email, EmailTempl
     public void resetPassword(PasswordReset passwordReset) {
         checkNotNull(passwordReset);
         
-        String cacheKey = passwordReset.getSptoken() + ":" + passwordReset.getStudyIdentifier();
-        String email = cacheProvider.getString(cacheKey);
+        // This pathway is unusual as the account may have an email address or a phone number, so test for both.
+        String emailCacheKey = passwordReset.getSptoken() + ":email:" + passwordReset.getStudyIdentifier();
+        String phoneCacheKey = passwordReset.getSptoken() + ":phone:" + passwordReset.getStudyIdentifier();
```

Comment:
This has the side effect that if someone requests a password reset before we deploy, and then we deploy, their password reset token is effectively invalidated. Is this an acceptable scenario?

Once potential workaround is to just "grandfather in" sptoken:study as implicitly meaning email, rather than trying to migrate the token.

Reply:
Yes good point. I've changed email cacheKey back to the original format, and added phone as a new cache key (which is being deployed with the phone changes). Should be backwards-compatible.